File: LATIHAN4DPBO2023-main/C++/Dosen.cpp
// Saya Raisyad Jullfikar NIM 2106238 mengerjakan soal Latihan 4 dalam mata kuliah Desain Pemrograman Berorientasi 
// Objek untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

#include <iostream>
using namespace std;

// Class Dosen yang merupakan anak dari class SivitasAkademik
// Merupakan class yang berisikan data - data Dosen pada umumnya
// Seperti Nip, pendter, keahlian, fakultas

class Dosen : public SivitasAkademik {
  // Deklarasi private attribute
  private:
    string __NIP;
    string __pendterakhir;
    string __keahlian;
    string __fakultas;

  // Constructor
  public:
    Dosen(string NIK = "", string name = "", string jenis_kelamin = "", string asal_universitas = "", string email_edu = "", string NIP = "", string pendterakhir = "", string keahlian = "", string fakultas = "") : SivitasAkademik(NIK, name, jenis_kelamin, asal_universitas, email_edu) {
      __NIP = NIP;
      __pendterakhir = pendterakhir;
      __keahlian = keahlian;
      __fakultas = fakultas;
    }

  // Setter and Getter
  public:
    string get_nip() { // Method untuk mengambil data NIM
      return this->__NIP;
    }

    void set_nip(string NIP) { // Method untuk menset atau mengubah atau menambah data NIM
      this->__NIP = NIP;
    }

    string get_pendterakhir() { // Method untuk mengambil data prodi
      return this->__pendterakhir;
    }

    void set_pendterakhir(string pendterakhir) { // Method untuk menset atau mengubah atau menambah data prodi
      this->__pendterakhir = pendterakhir;
    }

    string get_keahlian() { // Method untuk mengambil data prodi
      return this->__keahlian;
    }

    void set_keahlian(string keahlian) { // Method untuk menset atau mengubah atau menambah data prodi
      this->__keahlian = keahlian;
    }

    string get_fakultas() { // Method untuk mengambil data fakultas
      return this->__fakultas;
    }

    void set_fakultas(string fakultas) { // Method untuk menset atau mengubah atau menambah data fakultas
      this->__fakultas = fakultas;
    }

    ~Dosen() {} // Destructor
};

File: LATIHAN4DPBO2023-main/Python/Dosen.py
# Saya Raisyad Jullfikar NIM 2106238 mengerjakan soal Latihan 4 dalam mata kuliah Desain Pemrograman Berorientasi 
# Objek untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

# Import Class SivitasAkademik
from SivitasAkademik import *

# Class Dosen yang merupakan anak dari class SivitasAkademik
# Merupakan class yang berisikan data - data Dosen pada umumnya
# Seperti Nip, pendter, keahlian, fakultas

class Dosen(SivitasAkademik) :
  # Deklar private attr
  __NIP = ""
  __pendterakhir = ""
  __keahlian = ""
  __fakultas = ""
  
  # CONSTRUCTOR
  # Constructor kosong yang mengisi valuenya dengan default value atau string kosong
  def __init__ (self, NIK = "", name = "", jenis_kelamin = "", asal_universitas = "", email_edu = "", NIP = "", pendterakhir = "", keahlian = "", fakultas = "") :
    super().__init__(NIK,name,jenis_kelamin,asal_universitas,email_edu)
    # Fungsi super untuk memberikan akses dari kelas induknya yaitu Sivitas Akademik
    self.__NIP = NIP
    self.__pendterakhir = pendterakhir
    self.__keahlian = keahlian
    self.__fakultas = fakultas
  
  # SETTER AND GETTER
  
  def get_nip(self) : # Method untuk mengambil data NIM
    return self.__NIP
  def set_nip(self, NIP) : # Method untuk menset atau mengubah atau menambah data NIM
    self.__NIP = NIP;
  
  def get_pendterakhir(self) : # Method untuk mengambil data prodi
    return self.__pendterakhir
  def set_pendterakhir(self, pendterakhir) : # Method untuk menset atau mengubah atau menambah data prodi
    self.__pendterakhir = pendterakhir;

  def get_keahlian(self) : # Method untuk mengambil data prodi
    return self.__keahlian
  def set_keahlian(self, keahlian) : # Method untuk menset atau mengubah atau menambah data prodi
    self.__keahlian = keahlian;

  def get_fakultas(self) : # Method untuk mengambil data fakultas
    return self.__fakultas
  def set_fakultas(self, fakultas) : # Method untuk menset atau mengubah atau menambah data fakultas
    self.__fakultas = fakultas;